### ISSUE TITLE
[solved!]use-package error

### ISSUE THREAD

**Issue by bevsxyz** [tuh8888/chezmoi.el#19]
I am getting the following error when i try to run `(use-package chezmoi)`

```
Error (use-package): chezmoi/:catch: Loading file /usr/share/emacs/27.2/lisp/window.elc failed to provide feature ‘window’
```

Emacs version is 27.2

**Comment by tuh8888** [comment]
I'm not entirely sure how to help you with this one, as it's my impression that `window.el` should be provided by Emacs itself and has been for quite a while. I'm not getting that error on a fresh install, so it's going to be hard to debug without your particular computer. What OS are you running?

Could you try running emacs with no config: `emacs -Q`. Then try `(require 'window)` and tell me if that works.

You could also create a fork of chezmoi.el and remove https://github.com/tuh8888/chezmoi.el/blob/781783c483bc8fcdba3a230bb774c3a8a5ebe396/chezmoi.el#L51 . Then load that fork.

**Comment by bevsxyz** [comment]
I'm running Fedora 35. Will try loading window without the conf once I get to my computer.

**Comment by bevsxyz** [comment]
I get the following error

```elisp
Debugger entered--Lisp error: (invalid-function 'window)
  ('window)
  (progn ('window))
  eval((progn ('window)) t)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

**Comment by tuh8888** [comment]
That suggests to me that something is probably wrong with your Emacs installation and is unlikely to be a `chezmoi.el` issue. 

Do you have `/usr/share/emacs/27.2/lisp/window.elc` or `/usr/share/emacs/27.2/lisp/window.el` or any other of the `.el` files in `/usr/share/emacs/27.2/lisp/` (see https://github.com/emacs-mirror/emacs/tree/master/lisp for the list)?

If not, my recommendation is to try reinstalling Emacs.

**Comment by tuh8888** [comment]
I apologize for not having more ideas to help. You could try asking on an Emacs-related forum or Slack. Let me know if it does turn out to be a chezmoi.el problem though.

**Comment by bevsxyz** [comment]
I can load the window module via `M-x load-library RET window RET`. Apparently it errors only with require. Maybe it's an issue with my particular Emacs version.

**Comment by bevsxyz** [comment]
Solved!

I added `(provide 'window)` to the top of  `/usr/share/emacs/27.2/lisp/window.elc
` file. The file was not being recognised. Although compiled, I think it's pretty similar to minified javascript, therefore the trick worked. Probably will be solved by an emacs update. Until then this fix works like a charm. 🥳

**Comment by bevsxyz** [comment]
@tuh8888 Maybe it's a good idea to check whether it's a depreciated feature in the more upto date versions of Emacs.

**Comment by chrycheng** [comment]
Fixed in Emacs as of 28.0.90: https://github.com/emacs-mirror/emacs/commit/77aeddc1492e6b33b48366ca02d375088f754bd3
